src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs

```diff
@@ -77,24 +97,95 @@ public AdhocWorkspace CSharpWorkspace
             }
         }
 
-        public FormattingOptions Options { get; private set; } = null!;
-
-        public string NewLineString => Environment.NewLine;
-
-        public bool IsFormatOnType { get; private set; }
-
-        public bool AutomaticallyAddUsings { get; private set; }
-
-        public Range Range { get; private set; } = null!;
-
         /// <summary>A Dictionary of int (line number) to IndentationContext.</summary>
         /// <remarks>
         /// Don't use this to discover the indentation level you should have, use
         /// <see cref="TryGetIndentationLevel(int, out int)"/> which operates on the position rather than just the line.
         /// </remarks>
-        public IReadOnlyDictionary<int, IndentationContext> Indentations { get; private set; } = null!;
+        public IReadOnlyDictionary<int, IndentationContext> GetIndentations()
+        {
+            if (_indentations is null)
+            {
+                var sourceText = this.SourceText;
+                var indentations = new Dictionary<int, IndentationContext>();
 
-        public IReadOnlyList<FormattingSpan> FormattingSpans { get; private set; } = null!;
+                var previousIndentationLevel = 0;
+                for (var i = 0; i < sourceText.Lines.Count; i++)
+                {
+                    // Get first non-whitespace character position
+                    var nonWsPos = sourceText.Lines[i].GetFirstNonWhitespacePosition();
```

Comment:
```suggestion
                    var line = sourceText.Lines[i];
                    var nonWsPos = line.GetFirstNonWhitespacePosition();
```
And below.

Reply:
You should review with whitespace off, none of this changed, it just de-indented 😛